src/lib/Libifl/pbsD_msgjob.c

```diff
@@ -227,33 +228,28 @@ char *extend;
 		int i;
 		struct pbs_client_thread_connect_context *con;
 		char nd_ct_selstr[20];
-		char *endptr = NULL, *extend_dup = strdup(extend), *num_only;
-
-		if (extend_dup) {
-			num_only = strtok(extend_dup, " \'\"\n\t\v");
-			strtol(num_only, &endptr, 10);
-
-			if (num_only && !*endptr) {
-				snprintf(nd_ct_selstr, sizeof(nd_ct_selstr), "select=%s", num_only);
-				extend = nd_ct_selstr;
-			} else if ((i = set_resources(&attrib, extend, 1, &erp))) {
-				if (i > 1) {
-					snprintf(ebuff, sizeof(ebuff), "%s: %s\n", emsg_illegal_k_value, pbs_parse_err_msg(i));
-					emsg = strdup(ebuff);
-				} else
-					emsg = strdup("illegal -k value\n");
-				pbs_errno = PBSE_INVALSELECTRESC;
-			} else {
-				if (!attrib || strcmp(attrib->resource, "select")) {
-					emsg = strdup("only a \"select=\" string is valid in -k option\n");
-					pbs_errno = PBSE_IVALREQ;
-				} else
-					pbs_errno = PBSE_NONE;
-			}
-			free(extend_dup);
+		char *endptr = NULL;
+		long int rc_long;
+
+		errno = 0;
+		rc_long = strtol(extend, &endptr, 10);
+
+		if ((errno == 0) && (rc_long > 0)) {
```

Comment:
One minor suggestion, and then I'll be happy...
```suggestion
		if ((errno == 0) && (rc_long > 0) && (*endptr == '\0')) {
```

Reply:
> I think I could break this code using: -k '123 '
(Note there are spaces after 3.)
*endptr would not point to '\0' in this case.

This suggestion will fail to accept your initial breaking scenario `-k '123 '`. This is because `strtol()` ignores only leading white spaces and not trailing white space, so with this breaking input `*endptr` will not be `'\0'` but will be `' '`. Do you still want to commit this suggestion?

Reply:
Yes, it will fail to accept '123 ', and I think we want it to. You're testing whether the parameter is a number, and nothing more. If I supply 123abc as the argument to -k, then errno will be zero, rc_long will be 123, but *endptr will point to 'a'. Unless *endptr points to '\0', then the argument is not exclusively numeric. This will prevent people from trying to do anything out of the ordinary when supplying a numeric argument.

Reply:
One other thing I noticed was your check for rc_long > 0. That means negative numbers and zero will be identified as non-numeric. I think that is fine, since it will eventually throw an error.

Reply:
🤕 ,Now I caught the confusion. when you meant 

> I could break this code

 I thought you were providing a valid input which the code won't accept as valid. Now I can say that the initial code was never broken!. Anyways with the confusion cleared, I accept the suggestion.